lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2809,6 +2809,16 @@ uint32_t TypeSystemSwiftTypeRef::GetNumFields(opaque_compiler_type_t type,
           auto impl = [&]() -> llvm::Optional<uint32_t> {
             if (!type)
               return 0;
+            if (auto clang_type = GetAsClangTypeOrNull(type)) {
+              switch (clang_type.GetTypeClass()) {
+              case lldb::eTypeClassObjCInterface:
+              case lldb::eTypeClassObjCObject:
+                // Imported ObjC types are treated as having no fields.
+                return 0;
+              default:
+                return clang_type.GetNumFields();
```

Comment:
I suppose the validation macro should tell us if these assumptions are wrong?

Reply:
Yes, they did their job when I wrote this. At first I tried returning `0` for all clang types, but that broke some tests and allowed me to tweak this logic.

Reply:
We already talked about this offline, but in case anyone else is reading this — it would be good to understand why this doesn't already work via the GetSwiftRuntimeTypeInfo -> LLDBTypeInfoProvider code path.

Reply:
@adrian-prantl  The typeref for imported objc classes is `ObjCClassTypeRef`, ex:

```
(swift::reflection::ObjCClassTypeRef *) tr = 0x00006000029d5a10 {
  swift::reflection::TypeRef = (Kind = ObjCClass)
  Name = "NSApplication"
}
```

However, the TypeInfoProvider is currently used from `NominalTypeRef` and `BoundGenericTypeRef`.

See `visitAnyNominalTypeRef` https://github.com/apple/swift/blob/022311e438e7703c46e400ba01882ccb5a1236ca/stdlib/public/RemoteInspection/TypeLowering.cpp#L2302-L2379.